util/install.py

```diff
@@ -0,0 +1,130 @@
+import argparse
+import os
+import sys
+
+from utils import supported_distributions, identify_distribution, sh
+
+QuaggaVersion = "1.2.4"
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Install IPMininet with its dependencies")
+    parser.add_argument("-o", "--output-dir", help="Path to the directory that will store the dependencies",
+                        default=os.environ["HOME"])
+    parser.add_argument("-i", "--install-ipmininet", help="Install IPMininet", action="store_true")
+    parser.add_argument("-m", "--install-mininet", help="Install the last version of mininet and its dependencies",
+                        action="store_true")
+    parser.add_argument("-a", "--all", help="Install all daemons", action="store_true")
+    parser.add_argument("-q", "--install-quagga", help="Install Quagga (version %s) daemons" % QuaggaVersion,
+                        action="store_true")
+    parser.add_argument("-r", "--install-radvd", help="Install the RADVD daemon",
+                        action="store_true")
+    parser.add_argument("-s", "--install-sshd", help="Install the OpenSSH server", action="store_true")
+    return parser.parse_args()
+
+
+def install_mininet():
+    dist.install("git")
+
+    if dist.NAME == "Fedora":
+        mininet_opts = "-fnp"
+        dist.install("openvswitch", "openvswitch-devel", "openvswitch-test")
+        sh("systemctl enable openvswitch")
+        sh("systemctl start openvswitch")
+    else:
+        mininet_opts = "-a"
+
+    sh("git clone https://github.com/mininet/mininet.git",
+       "mininet/util/install.sh %s -s ." % mininet_opts,
+       "pip2 -q install mininet/",
+       cwd=args.output_dir)
+
+
+def install_quagga():
+    dist.install("autoconf", "automake", "libtool", "make", "gcc", "gawk", "pkg-config")
+
+    if dist.NAME == "Ubuntu" or dist.NAME == "Debian":
+        dist.install("libreadline-dev", "libc-ares-dev")
+    elif dist.NAME == "Fedora":
+        dist.install("readline-devel", "c-ares-devel")
+
+    quagga_src = os.path.join(args.output_dir, "quagga-%s" % QuaggaVersion)
+    quagga_tar = quagga_src + ".tar.gz"
+    sh("wget http://download.savannah.gnu.org/releases/quagga/quagga-%s.tar.gz" % QuaggaVersion,
+       "tar -zxvf '%s'" % quagga_tar,
+       cwd=args.output_dir)
+
+    quagga_install = os.path.join(args.output_dir, "quagga")
+    sh("./configure '--prefix=%s'" % quagga_install,
+       "make",
+       "make install",
+       cwd=quagga_src)
+
+    sh("rm -r '%s' '%s'" % (quagga_src, quagga_tar))
+
+    sh("groupadd quagga", may_fail=True)
+    sh("usermod -a -G quagga root", may_fail=True)
```

Comment:
We could follow mininet and create a dedicated `mininet` user

Reply:
I am not convinced of the added value of a new user.
IPMininet scripts will have to be run by the root user, not the ``mininet`` one.

Reply:
I agree, just for convenience. However, the mininet vm doesn't have sudo though.

Reply:
Quagga recommends to use another user as the daemon will run and expose itself to the network. This is not our primary use-case hence we do not that (+ this would require a bunch of fixes to ensures that all configs are accessible by that new quagga user etc).

Shouldn't we anyway move away from Quagga to FRR at some point?

Reply:
> Shouldn't we anyway move away from Quagga to FRR at some point?

I could change it by FRRouting (I already tested it) but this would first require the modification of the ipmininet BGP template because of this [issue](https://github.com/FRRouting/frr/issues/1644).